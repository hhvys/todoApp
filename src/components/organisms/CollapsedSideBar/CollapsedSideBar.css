.div__collapsed__sidebar {
    width: 42px;
    /*overflow: auto;*/
}

.div__collapsed__sidebar .list-toggle {
    min-height: 45px;
    height: 45px;
    background-color: #5B795A;
}

.div__collapsed__sidebar .tab-list {
    background-color: #F7F7F7;
    flex: 1;
}

    .div__collapsed__sidebar .tab-list div {
        height: 38px;
    }

.div__collapsed__sidebar .create-list {
    background-color: #F7F7F7;
    height: 42px;
    border-top: 1px solid #e0e0df;
}