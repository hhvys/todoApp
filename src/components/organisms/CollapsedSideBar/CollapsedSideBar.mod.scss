@import "../../../constants.mod";

.collapsedSidebar {
  width: $collpased-side-bar-width+px;
  height: 100vh;
}

.listToggle {
  min-height: $header-height+px;
  height: $header-height+px;
  background-color: #5B795A;
  cursor: pointer;
}

.tabList {
  background-color: $white-background;
  flex: 1;
}

.tabList div {
  height: 38px;
}

.createList {
  background-color: $white-background;
  height: $tab-height+px;
  border-top: 1px solid $border-color;
}

.icon {
  fill: $side-bar-primary;
  cursor: pointer;
}
