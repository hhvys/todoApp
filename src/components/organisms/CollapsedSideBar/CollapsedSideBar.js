import React from 'react';
import './CollapsedSideBar.css';
import Symbol from "../../atoms/logos/Symbol";
import Button from "../../atoms/Button/Button";
import CheckBox from "../../logos/CheckBox";

const container = [
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		// footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		active: true,
		footerContent: 4
	},
	{
		onClick: () => console.log('clicked'),
		onHeaderSymbolClick: () => console.log('headerClicked'),
		onFooterSymbolClick: () => console.log('footerClicked'),
		headerSymbol: 'listSymbol',
		footerSymbol: 'pencil',
		mainContent: 'tab1',
		style: {height: 38},
		footerContent: 4
	},
];


class CollapsedSideBar extends React.Component {

	renderListSymbols = (container) => {
		return (
			container.map(tab => {
				const {
					headerSymbol,
					style
				} = tab;
				return (
					<div className={"full-width d-flex align-items-center justify-content-center"} style={style}>
						<Symbol symbolType={headerSymbol} style={{fill: '#b9b9b9'}}/>
					</div>
				);
			})

		);
	};

	render() {
		const {className, ...props} = this.props;
		return (
			<div
				className={`div__collapsed__sidebar ${className ? className : ''} fixed d-flex flex-column align-items-center justify-content-start`}
				{...props}>
				<div className={"list-toggle full-width d-flex align-items-center justify-content-center"}>
					<Symbol symbolType={"listToggle"}/>
				</div>

				<div className={"tab-list full-width d-flex flex-column align-items-center justify-content-start"}>
					<div className={"full-width d-flex align-items-center justify-content-center"}>
						<Symbol symbolType={"threeDots"}/>
					</div>
					<div className={"full-width d-flex align-items-center justify-content-center"}>
						<Symbol symbolType={"threeDots"}/>
					</div>
				</div>

				<div className={"create-list full-width d-flex align-items-center justify-content-center"}>
					<Symbol symbolType={"plus"} style={{fill: '#328ad6'}}/>
				</div>
			</div>
		);
	}
}

export default CollapsedSideBar;