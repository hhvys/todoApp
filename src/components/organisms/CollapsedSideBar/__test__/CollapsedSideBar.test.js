import React from 'react';
import CollapsedSideBar from "../CollapsedSideBar";

describe('component', () => {
	describe('CollapsedSideBar', () => {
		it('should render', () => {
			expect(toJSON(shallow(<CollapsedSideBar/>))).toMatchSnapshot();
		});

		it('should pass collapseSideBar as onClick prop to Icons', () => {
			const prop = {
				collapseSideBar: jest.fn()
			};
			expect(toJSON(shallow(<CollapsedSideBar {...prop}/>))).toMatchSnapshot();
		});

	});
});
