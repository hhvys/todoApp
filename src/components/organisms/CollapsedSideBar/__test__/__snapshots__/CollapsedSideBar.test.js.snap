// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component CollapsedSideBar should pass collapseSideBar as onClick prop to Icons 1`] = `
<div
  className="collapsedSidebar  fixed d-flex flex-column align-items-center justify-content-start"
>
  <div
    className="listToggle full-width d-flex align-items-center justify-content-center"
  >
    <Icon
      height={20}
      iconType="LIST_TOGGLE"
      onClick={[Function]}
      width={20}
    />
  </div>
  <div
    className="tabList full-width d-flex flex-column align-items-center justify-content-start"
  >
    <div
      className="full-width d-flex align-items-center justify-content-center"
    >
      <Icon
        className="icon"
        height={20}
        iconType="INBOX"
        onClick={[Function]}
        width={20}
      />
    </div>
    <div
      className="full-width d-flex align-items-center justify-content-center"
    >
      <Icon
        className="icon"
        height={20}
        iconType="THREE_DOTS"
        onClick={[Function]}
        width={20}
      />
    </div>
  </div>
  <div
    className="createList full-width d-flex align-items-center justify-content-center"
  >
    <Icon
      className="icon"
      height={20}
      iconType="PLUS"
      onClick={[Function]}
      width={20}
    />
  </div>
</div>
`;

exports[`component CollapsedSideBar should render 1`] = `
<div
  className="collapsedSidebar  fixed d-flex flex-column align-items-center justify-content-start"
>
  <div
    className="listToggle full-width d-flex align-items-center justify-content-center"
  >
    <Icon
      height={20}
      iconType="LIST_TOGGLE"
      width={20}
    />
  </div>
  <div
    className="tabList full-width d-flex flex-column align-items-center justify-content-start"
  >
    <div
      className="full-width d-flex align-items-center justify-content-center"
    >
      <Icon
        className="icon"
        height={20}
        iconType="INBOX"
        width={20}
      />
    </div>
    <div
      className="full-width d-flex align-items-center justify-content-center"
    >
      <Icon
        className="icon"
        height={20}
        iconType="THREE_DOTS"
        width={20}
      />
    </div>
  </div>
  <div
    className="createList full-width d-flex align-items-center justify-content-center"
  >
    <Icon
      className="icon"
      height={20}
      iconType="PLUS"
      width={20}
    />
  </div>
</div>
`;
